modules/notification/indexer/indexer.go

```diff
@@ -126,7 +125,7 @@ func (r *indexerNotifier) NotifyMigrateRepository(ctx context.Context, doer, u *
 }
 
 func (r *indexerNotifier) NotifyPushCommits(ctx context.Context, pusher *user_model.User, repo *repo_model.Repository, opts *repository.PushUpdateOptions, commits *repository.PushCommits) {
-	if setting.Indexer.RepoIndexerEnabled && opts.RefFullName == git.BranchPrefix+repo.DefaultBranch {
+	if setting.Indexer.RepoIndexerEnabled && opts.RefFullName.ShortName() == repo.DefaultBranch {
```

Comment:
Such change doesn't seem right.

What if a tag's short name is also repo.DefaultBranch ?

(I would pause reviewing here, until these pending comments get resolved ....)

Reply:
Fixed in 5b31c36f20b746eb7d2ff6f7636c63620dab7857

Reply:
There would be more such abuses in the future.

I have objection for the ShortName, I think it's a improper design.

Reply:
How about a method named `BranchEqual`?